autonomx/WolframCA.cpp

```diff
@@ -34,111 +34,139 @@ WolframCA::WolframCA(int id, GeneratorMeta * meta) : Generator(id, meta){
 
 }
 
-WolframCA::~WolframCA()
-{
+WolframCA::~WolframCA() {
 
 }
 
-void WolframCA::initialize(){
+void WolframCA::initialize() {
 
-    if(flagDebug) {
-        std::chrono::nanoseconds now = std::chrono::duration_cast<std::chrono::nanoseconds>(
-                    std::chrono::system_clock::now().time_since_epoch()
-        );
+   if(flagDebug) {
+       std::chrono::nanoseconds now = std::chrono::duration_cast<std::chrono::nanoseconds>(
+                   std::chrono::system_clock::now().time_since_epoch());
 
-        qDebug() << "initialize:\t\t\tt = " << now.count() << "\tid = " << QThread::currentThreadId();
-    }
+       qDebug() << "initialize:\t\t\tt = " << now.count() << "\tid = " << QThread::currentThreadId();
+   }
 
-    // re-init algorithm here
+   // re-init algorithm here
 
-    // resize cells vector for current lattice size
-    cells.resize(latticeHeight * latticeWidth);
+   // resize cells vector for current lattice size
+   cells.resize(latticeHeight * latticeWidth);
 
-    // initialize cell values with only one cell in the middle
-   if (getRandSeed()==0){
-   for(int i = 0; i < (latticeHeight * latticeWidth); ++i) {
-        cells[i]=0;
-    }
-    cells[latticeWidth/2]=1;
+   // reset all cells to 0 / black
+   for(int i = 0; i < (latticeHeight * latticeWidth); i++) {
+       cells[i] = 0;
    }
-else{
-    //initialize cells with a random seed
-    for(int i = 0; i < (latticeHeight * latticeWidth); ++i) {
-        if (i>latticeWidth)
-        cells[i]=0;
-        // generate a random number magic b/w 0-1 and initialize to 1 if magic>0.5 o/w magic -> 0
-        else{
-            double magic = (float) rand()/RAND_MAX;
-            if (magic>getRandSeed())
-             cells[i] = 1;
-            else
-                cells[i]=0;
-        }
-    }
+
+   bool seedChosen = false;
+   double magic;
+
+   // user choice: random starting seed or no?
+   // if no random starting seed (default choice)
+   if (getRandSeed() == 0) {
+       // middle cell on first line is starting seed
+       cells[latticeWidth / 2] = 1;
+   }
+
+   // if random starting seed is selected by user
+   // TODO:
+   // random seed bug
+   // see below for temp fix and issue
+   else {
+       // initialize a random cell as starting cell
+       for(int i = 0; i < (latticeHeight * latticeWidth); i++) {
+           if (i >= latticeWidth) {
+               cells[i] = 0;
+           }
+           else {
+               // generate a random number magic b/w 0-1 and initialize to 1 if magic>0.5 o/w magic -> 0
+                magic = (float) rand() / RAND_MAX;
+               if (magic > getRandSeed()) {
+                   cells[i] = 1;
+                   seedChosen = true;
+               }
+               else {
+                   cells[i] = 0;
+               }
+           }
+       }
+
+       // TODO:
+       // fix this random seed bug
```

Comment:
Which random seed bug?

Reply:
We discussed it in the sprint review. It's working with a temporary fix but not as intended.